tests/foreman/api/test_errata.py

```diff
@@ -53,866 +53,896 @@
 from robottelo.helpers import add_remote_execution_ssh_key
 from robottelo.products import RepositoryCollection
 from robottelo.products import YumRepository
-from robottelo.test import APITestCase
 from robottelo.vm import VirtualMachine
 
 CUSTOM_REPO_URL = FAKE_9_YUM_REPO
 CUSTOM_REPO_ERRATA_ID = FAKE_2_ERRATA_ID
 
 
+@skip_if_not_set('clients', 'fake_manifest')
+@pytest.mark.skipif((not settings.repos_hosting_url), reason='Missing repos_hosting_url')
 @pytest.mark.run_in_one_thread
-class ErrataTestCase(APITestCase):
+class TestErrata:
```

Comment:
Is there any specific reason for keeping test cases outside of the "TestErrata" class?

Reply:
@devendra104 I do not understand the question.

Reply:
@JacobCallahan any advice on how to handle the above (what to do with decorators above a class)?

Reply:
Since they all need those decorators, you can do one of a couple of things.
1. Apply all of those decorators to each test.
2. Package those decorators into one, then apply that single decorator on each test.

Remember that decorators can be applied to a function without the syntactic sugar 🙂

Reply:
@JacobCallahan @swadeley Please use pytest's module scope mark list, there is a mechanism for this in place, there's no need to repeat it or create our own wrappers.

https://docs.pytest.org/en/stable/example/markers.html#marking-whole-classes-or-modules

```
# at the top of the test module
pytestmark = [pytest.mark.run_in_one_thread, pytest.mark.skipif(not settings.repos_hosting_url)]
```

Reply:
OK, that almost worked:
`Error evaluating 'skipif': you need to specify reason=STRING when using booleans as conditions.`

I used this:
`pytestmark = [pytest.mark.run_in_one_thread, pytest.mark.skipif((
    not settings.repos_hosting_url), reason='Missing repos_hosting_url')]`

Thank you

Reply:
@devendra104 The two classes have been removed (no point to have just one)